File: harness/semgrep/fixtures/c1-gated/verify.bad.py
# BAD (C1): on-chain proof only runs when signature is truthy; empty => bypass.


def verify_credential(cred):
    if cred.signature:
        verify_on_chain(cred.signature, cred.payload)
    return True


def verify_bare(signature, payload):
    if signature:
        check_on_chain_proof(signature, payload)
    return True

File: harness/semgrep/fixtures/c1-gated/verify.bad.ts
// BAD (C1): the on-chain proof only runs when `signature` is present, so an
// empty/omitted signature skips verification entirely => bypass.

interface Credential {
    signature?: string;
    payload: unknown;
}

async function verifyCredential(cred: Credential): Promise<boolean> {
    if (cred.signature) {
        await verifyOnChain(cred.signature, cred.payload);
    }
    // falls through as "accepted" when signature is absent
    return true;
}

async function verifyBare(signature: string | undefined, payload: unknown) {
    if (signature) {
        await checkOnChainProof(signature, payload);
    }
    return true;
}
